js/header-resizer.js

```diff
@@ -0,0 +1,63 @@
+/* global $ $$ dom */// dom.js
+/* global debounce */// toolbox.js
+/* global prefs */
+'use strict';
+
+(() => {
+  let curW, offset, active;
+  prefs.subscribe(dom.HWprefId, (key, val) => {
+    if (!active && val !== curW) {
+      getCurWidth();
+      setWidth(val);
+    }
+  });
+  $('#header-resizer').onmousedown = e => {
+    if (e.button) return;
+    getCurWidth();
+    offset = curW - e.clientX;
+    active = true;
+    document.body.classList.add('resizing-h');
+    document.on('mousemove', resize);
+    document.on('mouseup', resizeStop);
+  };
+
+  function getCurWidth() {
+    curW = parseFloat(document.documentElement.style.getPropertyValue('--header-width'))
+      || $('#header').offsetWidth;
+  }
+
+  /** @param {MouseEvent} e */
+  function resize(e) {
+    if (setWidth(offset + e.clientX)) {
+      debounce(save, 250, e.shiftKey);
+    }
+  }
+
+  function resizeStop() {
+    document.off('mouseup', resizeStop);
+    document.off('mousemove', resize);
+    document.body.classList.remove('resizing-h');
+    active = false;
+  }
+
+  function save(perPage) {
+    if (perPage) {
+      prefs.set(dom.HWprefId, curW);
+    } else {
+      for (const k of prefs.knownKeys) {
+        if (k.startsWith(dom.HW)) prefs.set(k, curW);
+      }
+    }
+  }
+
+  function setWidth(w) {
+    const delta = (w = dom.setHWProp(w)) - curW;
+    if (delta) {
+      curW = w;
+      for (const el of $$('.CodeMirror-linewidget[style*="width:"]')) {
```

Comment:
In the future, we may want to switch to `ResizeObserver` (probably in `codemirror-factory.js`) so it will always work not only when resizing the sidebar.

Reply:
In which case it'd be better? Also ResizeObserver is not implemented in our minimum supported browsers.

Reply:
> In which case it'd be better?

Pros after moving this CodeMirror code:

* The logic is decoupled from `header-resizer.js`.
* It is future-proof so that it can handle all kinds of editor resize.

It will be even better if this can be implemented in CodeMirror. Though I guess the dev would suggest us to refresh manually.

> ResizeObserver is not implemented in our minimum supported browsers.

Yeah that's the problem if we want to implement it now.

Reply:
I've tried extracting it and use MutationObserver on document.documentElement but I didn't like this solution. It's bigger, slower, more complicated compared to this trivial two-liner.